drivers/sensor/adxl345/adxl345.c

```diff
@@ -142,36 +142,36 @@ static int adxl345_init(struct device *dev)
 		return -ENODEV;
 	}
 
-	rc = i2c_reg_read_byte(dev, data->i2c_addr, ADXL345_DEVICE_ID_REG,
-			&dev_id);
+	rc = i2c_reg_read_byte(data->i2c_master, data->i2c_addr,
```

Comment:
How could this driver have ever worked if the wrong device was passed to i2c?

Reply:
This driver was developed and initially targetting the earlier version of Zephyr and this is a result of an error in refactoring.

The version that is currently on master unfortunately does not work , so this is a fix for that.